packages/js/client/src/__tests__/core/sanity.spec.ts

```diff
@@ -94,30 +114,190 @@ describe("sanity", () => {
       redirects: [
         {
           from: implementation1Uri,
-          to: implementation2Uri,
-        },
-      ],
+          to: implementation2Uri
+        }
+      ]
     });
 
     const redirects = client.getRedirects();
 
     expect(redirects).toEqual([
       {
         from: new Uri("wrap://ens/sha3.polywrap.eth"),
-        to: new Uri(defaultWrappers.sha3),
+        to: new Uri(defaultWrappers.sha3)
       },
       {
         from: new Uri("wrap://ens/uts46.polywrap.eth"),
-        to: new Uri(defaultWrappers.uts46),
+        to: new Uri(defaultWrappers.uts46)
       },
       {
         from: new Uri("wrap://ens/graph-node.polywrap.eth"),
-        to: new Uri(defaultWrappers.graphNode),
+        to: new Uri(defaultWrappers.graphNode)
       },
       {
         from: new Uri(implementation1Uri),
-        to: new Uri(implementation2Uri),
-      },
+        to: new Uri(implementation2Uri)
+      }
     ]);
   });
-});
+
+  test("validate requested uri is available", async () => {
+    const fooPath = `${__dirname}/../utils/validate/wrapper-a`;
+    const fooUri = `fs/${fooPath}/build`;
+    const greetingPath = `${__dirname}/../utils/validate/wrapper-b`;
+    const greetingUri = `fs/${greetingPath}/build`;
+    const ensGreetingPath = `${__dirname}/../utils/validate/wrapper-c`
+
+    process.env = {
+      ...process.env,
+      IPFS_GATEWAY_URI: providers.ipfs,
+      ENS_REG_ADDR: ensAddresses.ensAddress,
+      ENS_REGISTRAR_ADDR: ensAddresses.registrarAddress,
+      ENS_RESOLVER_ADDR: ensAddresses.resolverAddress,
+    };
+
+    await initTestEnvironment();
```

Comment:
To speed up the test, I think you could avoid using the test environment by deleting the build folders at the start of the test and then building the wrappers intermittently instead of adding resolvers. So you would only need to use the fs resolver and it would be added at the start. Instead of adding resolvers you would build the wrappers so that the build folders would exist. What do you think?

Reply:
this makes total sense, thanks :pray: